app/services/invites/accept_service.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Invites
+  class AcceptService < BaseService
+    def call(**args)
+      invite = Invite.find_by(token: args[:token], status: :pending)
+      return result.not_found_failure!(resource: 'invite') unless invite
+
+      ActiveRecord::Base.transaction do
+        result = UsersService.new.register_from_invite(
+          args[:email],
+          args[:password],
+          invite.organization_id,
+        )
+
+        invite.recipient = result.membership || Membership.find_by(user_id: result.user.id)
```

Comment:
What happens if at the end recipient is `nil`?

Reply:
well, it cannot really happen.
During invites the membership is created, and TBH that should be the only condition.
While copying the `register` method and adapting it for invites, I notices there is a condition if the user already exists then it only return the user, hence the `Membership.find_by`.

Does that sounds wrong ?

Reply:
I removed all code on the right of `||` and added a test for this case. Indeed result.membership should always exist